Increase the value of a kernel parameter to stop NVMe remounts

jsf9k opened the pull request:
## 🗣 Description

Setting the `nvme_core.io_timeout` kernel parameter to the highest allowable value fixes a bug where NVMe volumes occasionally fail and are mounted read-only.

See [this link](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/nvme-ebs-volumes.html#timeout-nvme-ebs-volumes) for more details.

## 💭 Motivation and Context

We have been seeing cases where NVMe volumes in AWS sporadically encounter a timeout condition and are remounted read-only.  This behavior interferes with the operation of the Cyber Hygiene commander, for example, and must cease.

## 🧪 Testing

I added a test to verify that the molecule code functions as intended.

TODO: I intend to deploy an image for each OS that we use and verify that the kernel parameter is set correctly.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Add template for /etc/sysctl.d config file, and a task to copy it over

This file will set the nvme_core.io_timeout kernel parameter to the
highest allowable value.  Setting this kernel parameter to a high
value fixes a bug where NVMe volumes occassionally fail and are
mounted read-only.

See this link for more details:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/nvme-ebs-volumes.html#timeout-nvme-ebs-volumes

jsf9k committed:
Use systemd-enabled Docker containers

systemd-sysctl is the service that actually applies the parameters
specified in the /etc/sysctl.d configuration files.

jsf9k committed:
Add a test to ensure that the /etc/sysctl.d config file is indeed copied over

dav3r reviewed (approved):
Looks good @jsf9k! Nice work. 👍 🎉

mcdonnnj commented on molecule/default/molecule.yml:
```diff
@@ -6,14 +6,37 @@ driver:
+  # Ansible currently insists on installing python2-dnf, which does
+  # not exist in Fedora 30.  Until that is resolved, we can't use
+  # Fedora 30.
+  # - name: fedora29_systemd
```
I think this should still reflect a Fedora 30 entry in the name and image even if it's not currently in use.

jsf9k replied:
Fixed in dea41ec.  You have sharp eyes!

mcdonnnj reviewed (changes requested):
This looks solid to me. Thanks for researching this and strong work 💪 Please see my one comment.

jsf9k committed:
Use udev instead of sysctl to set module options

Using sysctl won't work here since the module isn't until it is
actually needed.

jsf9k committed:
Fix two typos in some commented-out molecule configuration

Thanks to @mcdonnnj for noticing this error!

jsf9k committed:
Revert to the non-systemd images

Systemd is no longer needed since modprobe is configuring the module
parameters for us.

jsf9k committed:
Add prepare.yml playbook to install kmod

jsf9k commented:
I built a one-off CyHy Docker image and deployed it.  It looks like the `/etc/modprobe.d` approach is working great!

dav3r reviewed (approved):
Reviewed the latest changes and re-approved! 👍
